open-api/lang-tools/src/main/java/io/deephaven/lang/completion/CompletionRequest.java

```diff
@@ -79,11 +85,101 @@ public TableDefinition getTableDefinition(VariableProvider variables, String nam
+                    final ChunkerAssign check = itr.previous();
+                    result = findTableDefFromAssign(check);
+                    if (result != null) {
+                        break;
+                    }
+                }
+            }
+        }
         localDefs.put(name, result);
         return result;
     }
 
+    private TableDefinition findTableDefFromAssign(final ChunkerAssign check) {
+        final Node value = check.getValue();
+        if (value instanceof ChunkerInvoke) {
+            ChunkerInvoke invoke = (ChunkerInvoke) value;
+            if ("newTable".equals(invoke.getName())) {
+                // TODO: consider checking the scope of newTable... for now, too messy for ~0 value.
+                return convertNewTableInvocation(invoke);
+            }
+        }
+        return null;
+    }
+
+    private TableDefinition convertNewTableInvocation(final ChunkerInvoke invoke) {
+        final TableDefinition def = new TableDefinition();
+        final List<ColumnDefinition<?>> columns = new ArrayList<>();
+        for (Node argument : invoke.getArguments()) {
+            if (argument instanceof ChunkerInvoke) {
+                final ChunkerInvoke colInvoke = ((ChunkerInvoke) argument);
+                String colMethod = colInvoke.getName();
+                final List<Node> colArgs = colInvoke.getArguments();
+                final String colName = colArgs.isEmpty() ? null : toStringLiteral(colArgs.get(0));
+
+                switch (colMethod) {
+                    case "stringCol":
+                        columns.add(ColumnDefinition.fromGenericType(colName, String.class));
+                        break;
+                    case "dateTimeCol":
+                        columns.add(ColumnDefinition.fromGenericType(colName, DBDateTime.class));
+                        break;
+                    case "longCol":
+                        columns.add(ColumnDefinition.fromGenericType(colName, long.class));
+                        break;
+                    case "intCol":
+                        columns.add(ColumnDefinition.fromGenericType(colName, int.class));
+                        break;
+                    case "shortCol":
+                        columns.add(ColumnDefinition.fromGenericType(colName, short.class));
+                        break;
+                    case "byteCol":
+                        columns.add(ColumnDefinition.fromGenericType(colName, byte.class));
+                        break;
+                    case "charCol":
+                        columns.add(ColumnDefinition.fromGenericType(colName, char.class));
+                        break;
+                    case "doubleCol":
+                        columns.add(ColumnDefinition.fromGenericType(colName, double.class));
+                        break;
+                    case "floatCol":
+                        columns.add(ColumnDefinition.fromGenericType(colName, float.class));
+                        break;
```

Comment:
There should be specific methods off of ColumnDefinition for each primitive type.